pkg/intermediate/aggregate.go

```diff
@@ -338,6 +338,13 @@ func (a *AggregationProcess) addOrUpdateRecordInMap(flowKey *FlowKey, record ent
 	} else {
 		klog.Warning("FlowType does not exist in current record.")
 	}
+
+	// Ignore records from the Egress node for Pod-to-External flows
+	// if the Egress node is not the same as the Source node.
+	if isRecordFromEgressNode(flowType, record) {
```

Comment:
why don't we skip sending it from the Agent instead?

Reply:
Thanks Antonin bringing it up, I misunderstood the words from Srikar 'at Egress node' as 'from Egress node.' I agree skip sending it from the Agent is a better approach.